### PR TITLE
5401 add terserutil clear by fhirpath

### PR DESCRIPTION
**Pull request by fil512**
Previously, it was only possible to clear a top-level field on a resource using TerserUtil.  A new method has been added
to TerserUtil to support clearing a value by FhirPath.

### CODE REVIEW

File: hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/util/TerserUtilTest.java [hunk 1]
```diff
@@ -525,6 +529,38 @@ public void testClearFields() {
...
+		a1.setCity("Test");
+		p1.addAddress(a1);
+
+		Address a2 = new Address();
+		a2.addLine("Line 1");
+		a2.addLine("Line 2");
+		a2.setCity("Test");
+		p1.addAddress(a2);
+
+		TerserUtil.clearFieldByFhirPath(ourFhirContext, p1, "address.line");
```
Comment (volodymyr-korzh) [inline hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/util/TerserUtilTest.java:555]: Suggestion: Adding one more test case with deeper FHIR path. Example: `contact.address.line`

**Review by volodymyr-korzh — APPROVED** [review]

File: hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/util/TerserUtilTest.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/util/TerserUtilTest.java:555]: Good call.  You helped me catch an edge case I had not properly tested.
